Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -10,7 +11,90 @@ public Bishop(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var moves = Enumerable.Empty<Square>();
+            var piece = board.FindPiece(this);
+
+            for (var i = 1; i < 8; i++)
+            {
+                var row = piece.Row + i;
+                var col = piece.Col + i;
+                if (row >= 0 && row <= 7 && col >= 0 && col <= 7)
```

Comment:
tbh, I would probably change the board.GetPiece() method to return null if the piece is out of bounds, and get rid of this check